File: Python/1.19/1.19.due/caesar/caesar.py
# Paola Frunzio
# 1.17.2021
# caesar cipher

from sys import argv
from cs50 import get_string

if len(argv) != 2:
    print("Please enter the right command.")
    exit(1)

# get text to encrypt
plaintext = get_string("plaintext: ")

print("ciphertext: ")
k = int(argv[1])

# encrypting
for i in plaintext:
    if str.isalpha(i) == True:
        p = ord(i)
        p += k
        # to make sure it stays in the letter range of ascii
        while p >= 123:
            p -= 26
        print(chr(p), end="")
    else:
        print(i, end="")
print()

File: C/11.11/caesar.c
//Paola Frunzio
//11.10-11.2020
//caesar cipher

#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Please enter the right command.\n");
        return 1;
    }

    int i, n, p, k = atoi(argv[1]);

    string plaintext = get_string("plaintext: ");

    printf("ciphertext: ");

    //encrypting
    for (i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (isalpha(plaintext[i]) != 0)
        {
            p = (int)(plaintext[i]);
            p += k;
            //to make sure it stays a letter
            while (p >= 123)
            {
                p -= 26;
            }
            printf("%c", (char)p);
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
}